app/utils/service_client/middleware/param_encoder.rb

```diff
@@ -0,0 +1,27 @@
+module ServiceClient
+  module Middleware
+
+    class ParamEncoder < MiddlewareBase
+      def enter(ctx)
+        params = ctx[:req][:params]
+        ctx[:req][:params] = encode(params)
+        ctx
+      end
+
+      private
+
+      def encode(params)
```

Comment:
Thanks for a great suggestion! The `is_a?` felt awkward but I didn't know any better. Your suggestion sounds good, I'll change to that.